crates/primitives/src/sorted_vec.rs

```diff
@@ -41,12 +41,26 @@ impl<T: Ord + Clone> SortedVec<T> {
 
     /// Checks if the [`SortedVec`] contains the given value.
     pub fn contains(&self, value: &T) -> bool {
-        self.binary_search(value).is_ok()
+        self.binary_search(value).is_some()
     }
 
     /// Perform binary search on the vector
-    pub fn binary_search(&self, value: &T) -> Result<usize, usize> {
-        self.inner.binary_search(value)
+    pub fn binary_search(&self, value: &T) -> Option<&T> {
+        self.inner
+            .binary_search(value)
+            .ok()
+            .map(|idx| &self.inner[idx])
+    }
+
+    pub fn binary_search_by_key<F, B>(&self, b: &B, mut f: F) -> Option<&T>
+    where
+        F: FnMut(&T) -> &B,
+        B: Ord,
+    {
+        self.inner
+            .binary_search_by(|k| f(k).cmp(b))
+            .ok()
+            .map(|idx| &self.inner[idx])
```

Comment:
I am a little bit perplexed as to why this doesn't also call out to `.binary_search_by_key`.